fix: remove version patches and add MM webhooks

siggiskulason opened the pull request:
device-camera-go and device-rest-go do not need to be patched as
the upstream repositories have been fixed.
Also added MM messages on sync and LP failures.

tonyespy commented on .github/workflows/wf-device-camera.yml:
```diff
@@ -6,13 +6,16 @@ jobs:
-          canonical-repo: "edgex-device-camera-go"
-          patch-file: "0001-snapcraft-version-patch.diff"
-          ssh-private-key: ${{ secrets.SSH_CANONICAL_EDGEX_DEVICE_CAMERA }}
+    - name: Build and version device-camera
```
Should version still be included in the name?

siggiskulason replied:
In this case, "version" refers to the process of creating the VERSION file, which is still done there - it's just the patching of the versioning logic which has been removed.

tonyespy commented on .github/workflows/wf-device-camera.yml:
```diff
@@ -43,10 +46,14 @@ jobs:
-          consumer_name: ${{ secrets.LP_CONSUMER_NAME }}
-          access_token: ${{ secrets.LP_ACCESS_TOKEN }}
-          access_secret: ${{ secrets.LP_ACCESS_SECRET }}
+    - name: Kick off Launchpad build
```
Looks like the formatting was updated too, was this intentional?

siggiskulason replied:
It was - It wasn't uniform across the three steps so I cleaned that up. Perhaps that should have been left out of this commit though as it's not related to the change which is actually being made.

tonyespy commented on .github/workflows/wf-device-rest.yml:
```diff
@@ -6,13 +6,16 @@ jobs:
-          canonical-repo: "edgex-device-rest-go"
-          ssh-private-key: ${{ secrets.SSH_CANONICAL_EDGEX_DEVICE_REST }}  
-          patch-file: "0001-snapcraft-version-patch.diff" 
+    - name: Sync and version repo
```
Same as previous comment...

tonyespy commented on .github/workflows/wf-device-rest.yml:
```diff
@@ -43,10 +46,14 @@ jobs:
-          consumer_name: ${{ secrets.LP_CONSUMER_NAME }}
-          access_token: ${{ secrets.LP_ACCESS_TOKEN }}
-          access_secret: ${{ secrets.LP_ACCESS_SECRET }}         
+    - name: Kick off Launchpad build
```
formatting?

tonyespy reviewed (commented):
A couple of comments inline...

Also, I think it probably makes sense to split into two commits, one for camera, and one for rest.

tonyespy commented:
Also, two other comments:

 - since commits should be logical changes you might also want to break out the change for each which drops the version patch and the change that adds the MM alert
 - and finally, in the commit summary, maybe change "patch" to "version patch"?

siggiskulason commented:
Thanks @tonyespy  - Yes, this PR did sort of break the cardinal rule of making distinct, logical changes. It is really making three changes
1) minor changes to tabs to make the indentation standard across the steps
2) changes to remove the reference to a patch file - and to delete that patch file, for both device-camera and device-rest
3) adding the MM webhook.

Should I rather then create four PRs:
1) removing device-camera-go patch reference / and delete that patch file
2) removing device-rest-go patch reference / and delete that patch file
3) add webhooks in device-camera
4) add webhooks in device-rest

That's fine with me - I just didn't want to overburden you with reviewing/merging PRs for relatively trivial changes to the pipeline :-)

Commit:
remove unused version patch files

device-camera-go and device-rest-go no longer need to be patched so the two patch files
can be removed safely.

Commit:
remove reference to patch file in device-camera workflow

device-camera versioning has been updated in upstream edgexfoundry repository and
we should no longer apply the versioning patch in our workflow.

Commit:
remove reference to patch file in device-rest workflow

device-rest versioning has been updated in upstream edgexfoundry repository and
we should no longer apply the versioning patch in our workflow.

Commit:
add MM messages on camera workflow failure

Added webhooks to send messages when the initial sync step or the launchpad build step fails.

Commit:
add MM messages in device-rest workflow

Added webhooks to send messages when the initial sync step or the launchpad build step fails.

tonyespy reviewed (approved):
LGTM
